### PR TITLE
XML string array errors feedback

### PR DESCRIPTION
**Pull request by MelReyCG**
This PR aims to improve the error messages from XML string array in a more user-friendly way:
- It adds more context to the parsing exceptions so the user can retrieve the error cause faster,
- In order to throw these exception properly, a modification of `LvArray::input::stringToArray()` is needed (LvArray PR [#280](https://github.com/GEOSX/LvArray/pull/280)),
- It modifies the generated schema.xsd so these errors can be detected by xml validation,

This PR solves the case 1 of issue #2320.
LvArray PR [#280](https://github.com/GEOSX/LvArray/pull/280)

### CODE REVIEW

**Review by untereiner — APPROVED** [review]

File: src/coreComponents/common/DataTypes.hpp [hunk 1]
```diff
@@ -643,8 +643,17 @@ class rtTypes
...
+    string rs = "[^,\\{\\}\\s]+\\s*";
+
+    // Regex to match a string that does not contain any whitespaces nor the characters ,{}
+    string rse = "[^,\\{\\}\\s]*\\s*";
+
+    // Regex to match a path: a string that can't be empty and does not contain any space nor the characters *?<>|:",
+    string rp = "[^*?<>\\|:\";,\\s]+\\s*";//"[^*?\\&lt;\\&gt;\\|:\\&quot;,\\s]+\\s*";
+
+    // Regex to match a path: a string that does not contain any space nor the characters *?<>|:",
+    string rpe = "[^*?<>\\|:\";,\\s]*\\s*";//"[^*\\?&lt;\\&gt;\\|:\\&quot;,\\s]*\\s*";
```
[the remark this answers is not in the document]
Comment (MelReyCG) [reply src/coreComponents/common/DataTypes.hpp:656]: Sure!

[the remark this answers is not in the document]
Comment (MelReyCG) [reply src/coreComponents/common/DataTypes.hpp:656]: I'm afraid we cannot because of r1 and r2s need string operations. We would need string_views here.

**Review by cssherman — APPROVED** [review]

File: src/coreComponents/common/DataTypes.hpp [hunk 2]
```diff
@@ -641,15 +643,31 @@ class rtTypes
...
     // ([eE][-+]?[\\d]+|\\s*)  matches an optional scientific notation number
     // Note: the xsd regex implementation does not allow an empty branch, so use allow whitespace at the end
+    // TODO c++17: Move to static constexpr std::string_view
     string rr = "[+-]?[\\d]*([\\d]\\.?|\\.[\\d])[\\d]*([eE][-+]?[\\d]+|\\s*)";
 
-    // Regex to match a string that does not contain the characters  ,{}
-    string rs = "[^,\\{\\}]*";
+    // Regex to match a string that can't be empty and does not contain any whitespaces nor the characters ,{}
+    // TODO c++17: Move to static constexpr std::string_view
+    string rs = "[^,\\{\\}\\s]+\\s*";
```
Comment (TotoGaz) [inline src/coreComponents/common/DataTypes.hpp:651]: Do you need to take action with potential initial spaces (like ` toto`) or is it OK?

**Review by TotoGaz — APPROVED** [review]
Great, thx @MelReyCG 
@cssherman Do you think there are any implications with those regex in our global process (python validation or anything else)?

File: src/coreComponents/common/DataTypes.hpp (hunk 2 above)
Comment (cssherman) [reply src/coreComponents/common/DataTypes.hpp:651]: This pattern will be permissive of spaces at the end of the string, but not at the end, which _should_ be fine, right?

```
"hello": match
"hello world": no match
"hello, world": no match
" hello": no match
"hello ": match
```

**Review by rrsettgast — APPROVED** [review]

**Pull request merged**